src/main.rs

```diff
@@ -214,6 +219,14 @@ fn real_main(flags: Flags, config: &Config) -> CliResult {
                        flags.flag_all);
             println!("");
         }
+    } else if flags.flag_json {
+        println!("{}", serde_json::to_string_pretty(
+            &graph_to_tree(package.package_id(),
```

Comment:
It seems a bit strange that this mode would ignore a bunch of the flags the other modes respect.